api/v1alpha1/httpchaos_types.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KindHTTPChaos is the kind for http chaos
+const KindHTTPChaos = "HTTPChaos"
+
+func init() {
+	all.register(KindHTTPChaos, &ChaosKind{
+		Chaos:     &HTTPChaos{},
+		ChaosList: &HTTPChaosList{},
+	})
+}
+
+// HTTPAction represents the chaos action about I/O action.
+type HTTPChaosAction string
+
+const (
+	HTTPDelayAction HTTPChaosAction = "delay"
+	HTTPAbortAction                 = "abort"
+	HTTPMixedAction                 = "mixed"
+)
+
+type Matcher struct {
+	Name           string  `json:"name"`
+	ExactMatch     *string `json:"exact_match,omitempty"`
+	RegexMatch     *string `json:"regex_match,omitempty"`
+	SafeRegexMatch *string `json:"safe_regex_match,omitempty"`
+	RangeMatch     *string `json:"range_match,omitempty"`
+	PresentMatch   *string `json:"present_match,omitempty"`
+	PrefixMatch    *string `json:"prefix_match,omitempty"`
+	SuffixMatch    *string `json:"suffix_match,omitempty"`
+	InvertMatch    *string `json:"invert_match,omitempty"`
+}
+
+type HTTPChaosSpec struct {
+	// Selector is used to select pods that are used to inject chaos action.
+	Selector SelectorSpec `json:"selector"`
+
+	// Scheduler defines some schedule rules to
+	// control the running time of the chaos experiment about pods.
+	Scheduler *SchedulerSpec `json:"scheduler,omitempty"`
+
+	// Action defines the specific pod chaos action.
+	// Supported action: delay | abort | mixed
+	// Default action: delay
+	// +kubebuilder:validation:Enum=delay;abort;mixed
+	Action HTTPChaosAction `json:"action"`
+
+	// Mode defines the mode to run chaos action.
+	// Supported mode: one
+	// +kubebuilder:validation:Enum=one
+	Mode PodMode `json:"mode"`
+
+	// Value is required when the mode is set to `FixedPodMode` / `FixedPercentPodMod` / `RandomMaxPercentPodMod`.
+	// If `FixedPodMode`, provide an integer of pods to do chaos action.
+	// If `FixedPercentPodMod`, provide a number from 0-100 to specify the percent of pods the server can do chaos action.
+	// IF `RandomMaxPercentPodMod`,  provide a number from 0-100 to specify the max percent of pods to do chaos action
+	// +optional
+	Value string `json:"value"`
```

Comment:
seems that these comment needs to modify, the `Mode` only support `one`

Reply:
I will change the `Mode` into common one. It support by common type .